donkeycar/parts/datastore_v2.py

```diff
@@ -14,11 +15,14 @@ class Seekable(object):
     This reader maintains an index of line lengths, so seeking a line is a O(1) operation.
     '''
 
-    def __init__(self, file, method='a+', line_lengths=list()):
-        self.method = method
+    def __init__(self, file, read_only=False, line_lengths=list()):
         self.line_lengths = list()
         self.cumulative_lengths = list()
+        self.method = 'r' if read_only else 'a+'
         self.file = open(file, self.method, newline=NEWLINE)
```

Comment:
We should probably have a `with` here...

Reply:
We cannot use a context manager here, given the  explicit `close()` semantics.

Reply:
Ok - I missed that.